lib/components/viewers/stop-viewer.js

```diff
@@ -240,35 +342,61 @@ class RouteRow extends Component {
 }
 
+const ONE_HOUR_IN_SECONDS = 3600
+
+function getHomeTime (homeTimezone) {
+  const now = moment()
+  return now.tz(homeTimezone).diff(now.clone().startOf('day'), 'seconds')
+}
+
 // helper method to generate stop time w/ status icon
-function getFormattedStopTime (stopTime, homeTimezone) {
+function getFormattedStopTime (stopTime, homeTimezone, arrivingText = 'Due') {
   const now = moment()
   const serviceDay = moment(stopTime.serviceDay * 1000)
-  const currentTime = now.diff(now.clone().startOf('day'), 'seconds')
-  const differentDay = moment().date() !== serviceDay.date()
-
-  const inHomeTimezone = homeTimezone &&
-    now.tz(homeTimezone).format('Z') === now.tz(moment.tz.guess()).format('Z')
+  const currentHomeTime = getHomeTime(homeTimezone)
+  const differentDay = now.date() !== serviceDay.date()
 
+  const userTimeZone = moment.tz.guess()
+  const inHomeTimezone = homeTimezone && homeTimezone === userTimeZone
   // Determine whether to show departure as countdown (e.g. "5 min") or as HH:MM time
-  const showCountdown = inHomeTimezone && !differentDay &&
-    stopTime.realtimeDeparture - currentTime < 3600 &&
-    stopTime.realtimeDeparture > currentTime
-
+  const secondsUntilArrival = stopTime.realtimeArrival - currentHomeTime
+  const arrivesInFuture = stopTime.realtimeArrival > currentHomeTime
+  const showCountdown = !differentDay &&
+    secondsUntilArrival < ONE_HOUR_IN_SECONDS &&
+    arrivesInFuture
+
+  // Use arriving test (e.g., Due) if vehicle is approaching.
+  const countdownString = secondsUntilArrival < 60
+    ? arrivingText
+    : formatDuration(secondsUntilArrival)
+  // Only show timezone (e.g., PDT) if user is not in home time zone (e.g., user
+  // in New York, but viewing a trip planner for service based in Los Angeles).
+  const tzToDisplay = inHomeTimezone ? null : homeTimezone
+  const formattedTime = formatStopTime(stopTime.realtimeArrival, tzToDisplay)
   return (
     <div>
       <div style={{ float: 'left' }}>
         {stopTime.realtimeState === 'UPDATED'
-          ? <Icon type='rss' style={{ color: '#888', fontSize: '0.8em', marginRight: 2 }} />
-          : <Icon type='clock-o' style={{ color: '#888', fontSize: '0.8em', marginRight: 2 }} />
+          ? <Icon
+            type='rss'
+            style={{ color: '#888', fontSize: '0.8em', marginRight: 2 }} />
+          : <Icon
+            type='clock-o'
+            style={{ color: '#888', fontSize: '0.8em', marginRight: 2 }} />
         }
       </div>
       <div style={{ marginLeft: 20, fontSize: differentDay ? 12 : 14 }}>
-        {differentDay && <div style={{ marginBottom: -4 }}>{serviceDay.format('dddd')}</div>}
-        <div>{showCountdown
-          ? formatDuration(stopTime.realtimeDeparture - currentTime)
-          : formatStopTime(stopTime.realtimeDeparture)
-        }</div>
+        {differentDay &&
+          <div style={{ marginBottom: -4 }}>{serviceDay.format('dddd')}</div>
+        }
+        <div>
+          {showCountdown
```

Comment:
Seems incorrect to not show a countdown for arrivals after midnight.

Reply:
See [previous comment](https://github.com/opentripplanner/otp-react-redux/pull/62/files#r297342085).